nxc/helpers/bloodhound.py

```diff
@@ -43,26 +43,26 @@ def add_user_bh(user, domain, logger, config):
         try:
             with driver.session() as session, session.begin_transaction() as tx:
                 for info in users_owned:
-                    distinguished_name = "".join(["DC=" + dc + "," for dc in info["domain"].split(".")]).rstrip(",")
-                    domain_query = tx.run(f'MATCH (d:Domain) WHERE d.distinguishedname STARTS WITH "{distinguished_name}" RETURN d').data()
+                    distinguished_name = "".join([f"DC={dc}," for dc in info["domain"].split(".")]).rstrip(",")
+                    domain_query = tx.run(f"MATCH (d:Domain) WHERE d.distinguishedname STARTS WITH '{distinguished_name}' RETURN d").data()
                     if not domain_query:
                         raise Exception("Domain not found in bloodhound")
                     else:
                         domain = domain_query[0]["d"].get("name")
 
                     if info["username"][-1] == "$":
-                        user_owned = info["username"][:-1] + "." + domain
+                        user_owned = f"{info['username'][:-1]}.{domain}"
                         account_type = "Computer"
                     else:
-                        user_owned = info["username"] + "@" + domain
+                        user_owned = f"{info['username']}@{domain}"
                         account_type = "User"
 
 
-                    result = tx.run(f'MATCH (c:{account_type} {{name:"{user_owned}"}}) RETURN c')
+                    result = tx.run(f"MATCH (c:{account_type} {{name:'{user_owned}''}}) RETURN c")
```

Comment:
It looks like an extra single quote got caught in here? I don't have bloodhound set up in my lab, have you tested it?

Reply:
Damn it, yes

Reply:
Interestingly it still worked